forge-gui/res/cardsfolder/upcoming/brasss_tunnel_grinder_tecutlan_the_searing_rift.txt

```diff
@@ -0,0 +1,26 @@
+Name:Brass's Tunnel-Grinder
+ManaCost:2 R
+Types:Legendary Artifact
+T:Mode$ ChangesZone | ValidCard$ Card.Self | Origin$ Any | Destination$ Battlefield | Execute$ TrigDiscard | TriggerDescription$ When CARDNAME enters the battlefield, discard any number of cards, then draw that many cards plus one.
+SVar:TrigDiscard:DB$ Discard | AnyNumber$ True | Optional$ True | Mode$ TgtChoose | RememberDiscarded$ True | SubAbility$ DBDraw
+SVar:DBDraw:DB$ Draw | Defined$ You | NumCards$ Remembered$Amount/Plus.1 | SubAbility$ DBCleanup
+SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+T:Mode$ Phase | Phase$ End of Turn | ValidPlayer$ You.descended | TriggerZones$ Battlefield | Execute$ TrigPutCounter | TriggerDescription$ At the beginning of your end step, if you descended this turn, put a bore counter on CARDNAME. Then if there are three or more bore counters on it, remove those counters and transform it. (You descended if a permanent card was put into your graveyard from anywhere.)
+SVar:TrigPutCounter:DB$ PutCounter | CounterType$ BORE | SubAbility$ DBBranch
+SVar:DBBranch:DB$ Branch | BranchConditionSVar$ Count$Valid Card.Self+counters_GE3_BORE | TrueSubAbility$ DBRemoveCounter
+SVar:DBRemoveCounter:DB$ RemoveCounter | CounterType$ BORE | CounterNum$ All | SubAbility$ DBTransform
+SVar:DBTransform:DB$ SetState | Defined$ Self | Mode$ Transform
+AlternateMode:DoubleFaced
+Oracle:When Brass's Tunnel-Grinder enters the battlefield, discard any number of cards, then draw that many cards plus one.\nAt the beginning of your end step, if you descended this turn, put a bore counter on Brass's Tunnel-Grinder. Then if there are three or more bore counters on it, remove those counters and transform it. (You descended if a permanent card was put into your graveyard from anywhere.)
+
+ALTERNATE
+
+Name:Tecutlan, the Searing Rift
+ManaCost:no cost
+Types:Legendary Land Cave
+A:AB$ Mana | Cost$ T | Produced$ R | TriggersWhenSpent$ SpentTrigger | SubAbility$ DBSpace | SpellDescription$ Add {R}.
+SVar:DBSpace:DB$ BlankLine | SpellDescription$ ,,,,,,Whenever you cast a permanent spell using mana produced by CARDNAME, discover X, where X is that spell's mana value.
```

Comment:
I think this needs more support, this is not the normal TriggersWhenSpent that's tied to its mana ability as delayed trigger

This is a trigger on its own with the consequence it can work for any other mana ability the card woud have available

Reply:
Seems to work in basic testing